include/AIToolbox/MDP/Model.hpp

```diff
@@ -327,16 +328,25 @@ namespace AIToolbox::MDP {
              */
             const RewardMatrix & getRewardFunction() const;
 
+            /**
+             * @brief This function returns the vector of the allowed actions in a given state.
+             *
+             * @return The vector of allowed actions.
+             */
+            template <typename ST = size_t, typename AT = size_t>
+            std::vector<AT> getAllowedActions(ST s) const;
```

Comment:
I would also prefer to have a separate MDP class to model variable action problems, rather than cramming this here. It makes much more sense to me to have a different MDP class with something like a `getA(size_t s)` function (which also has the added advantage of not having to allocate a vector every time it's called), which is specifically designed to handle that problem.

Reply:
100% with you on that, the allocation of the vector was bugging me as well. So something like a ```VariableModel``` then. I might combine it with the comment above on templating but it's doing something different. So maybe still having the ```BaseModel``` where the ```Model``` and the ```VariableModel``` would sit side by side in the hierarchy.

Reply:
`BaseModel` is not really needed. `VariableActionModel` could work, but again the thing is that no other method in the library would support it. For now for me it's fine if the library allows to introduce them to a later time (with the UCB_selects_the_action mechanism), but not sure whether we should actually put this in.

Reply:
See above.